packages/governance/src/question.js

```diff
@@ -0,0 +1,195 @@
+// @ts-check
+
+import { Far, passStyleOf } from '@agoric/marshal';
+import { sameStructure } from '@agoric/same-structure';
+import { makeHandle } from '@agoric/zoe/src/makeHandle.js';
+import { Nat } from '@agoric/nat';
+
+import { assertType, ParamType } from './paramManager.js';
+
+const { details: X, quote: q } = assert;
+
+// Topics being voted on are 'Questions'. Before a Question is known to a
+// registrar, the parameters can be described with a QuestionSpec. Once the
+// question has been presented to a Registrar, there is a QuestionDetails record
+// that also includes the VoteCounter which will determine the outcome and the
+// questionHandle that uniquely identifies it.
+
+/**
+ * "unranked" is more formally known as "approval" voting, but this is hard for
+ * people to intuit when there are only two alternatives.
+ *
+ * @type {{
+ *   UNRANKED: 'unranked',
+ *   ORDER: 'order',
+ * }}
+ */
+const ChoiceMethod = {
+  UNRANKED: 'unranked',
+  ORDER: 'order',
+};
+
+/** @type {{
+ *   PARAM_CHANGE: 'param_change',
+ *   ELECTION: 'election',
+ *   SURVEY: 'survey',
+ * }}
+ */
+const ElectionType = {
+  // A parameter is named, and a new value proposed
+  PARAM_CHANGE: 'param_change',
+  // choose one or multiple winners, depending on ChoiceMethod
+  ELECTION: 'election',
+  SURVEY: 'survey',
+};
+
+/** @type {{
+ *   MAJORITY: 'majority',
+ *   NO_QUORUM: 'no_quorum',
+ *   ALL: 'all',
+ * }}
+ */
+const QuorumRule = {
+  MAJORITY: 'majority',
+  NO_QUORUM: 'no_quorum',
+  // The election isn't valid unless all voters vote
+  ALL: 'all',
+};
+
+/** @param {SimpleIssue} issue */
```

Comment:
This style where the validation function parameter types are narrow sorta works against the tools we're using. As noted in [Feb 26 discussion](https://github.com/Agoric/agoric-sdk/issues/72#issuecomment-786837990), we get better support from the tools if we use:

```suggestion
/**
 * @param {unknown} issue
 * @returns { asserts issue is SimpleIssue }
 */
```

Then the tools tell us that we're not actually checking that `typeof issue === 'object'`. I'll make another suggestion below...

Reply:
done for all the looksLike... methods